exercise1/terraform/main.tf

```diff
@@ -0,0 +1,505 @@
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "443"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = azurerm_resource_group.ex1.name
+  network_security_group_name = azurerm_network_security_group.ex1_app_gw_netsecg.name
+}
+
+resource "azurerm_network_security_rule" "health_probe_inbound" {
+  name                        = "AllowHealthProbe"
+  priority                    = 1003
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "65200-65535"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = azurerm_resource_group.ex1.name
+  network_security_group_name = azurerm_network_security_group.ex1_app_gw_netsecg.name
+}
+
+# Deny all inbound traffic not explicitly allowed
+resource "azurerm_network_security_rule" "deny_all_inbound" {
+  name                        = "DenyAllInbound"
+  priority                    = 4096
+  direction                   = "Inbound"
+  access                      = "Deny"
+  protocol                    = "*"
+  source_port_range           = "*"
+  destination_port_range      = "*"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = azurerm_resource_group.ex1.name
+  network_security_group_name = azurerm_network_security_group.ex1_app_gw_netsecg.name
+}
+
+resource "azurerm_public_ip" "ex1_app_gw_pub_ip" {
+  name                = "${var.rg_name}_app_gw_pub_ip"
+  location            = azurerm_resource_group.ex1.location
+  resource_group_name = azurerm_resource_group.ex1.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+resource "azurerm_user_assigned_identity" "ex1_app_gw_ass_iden" {
+  name                = "${var.rg_name}_app_gw_ass_iden"
+  location            = azurerm_resource_group.ex1.location
+  resource_group_name = azurerm_resource_group.ex1.name
+}
+
+resource "azurerm_application_gateway" "ex1_app_gw" {
+  name                = "${var.rg_name}_app_gw"
+  resource_group_name = azurerm_resource_group.ex1.name
+  location            = azurerm_resource_group.ex1.location
+
+  sku {
+    name     = "Standard_v2"
+    tier     = "Standard_v2"
+    capacity = 1
+  }
+
+  gateway_ip_configuration {
+    name      = "${var.rg_name}_app_gw_ip_config"
+    subnet_id = azurerm_subnet.ex1_subnet_app_gw.id
+  }
+  # # Bonus attempt
+  # waf_configuration {
+  #   enabled          = true
+  #   firewall_mode    = "Detection"
+  #   rule_set_version = 3.2
+  # }
+  trusted_root_certificate {
+    name                = local.cert_tls_ssl
+    key_vault_secret_id = azurerm_key_vault_certificate.ex1_cert_appgw.secret_id
```

Comment:
This seems to be your SSL server certificate not the root CA. You need to pass the server cert instead.

Reply:
Taking a stab at it and going to test with ssl_certificate.

I am unsure if you mean that my trusted root cert block is wrong or that i am passing the wrong cert from the akv